### PR TITLE
docs(performance): add fine tuning docs

### PR DESCRIPTION
**Pull request by astencel-sumo**
Here's a minimal starting point for describing the fine tuning of the collector.

### CODE REVIEW

File: docs/Performance.md [hunk 1]
```diff
@@ -69,3 +69,41 @@ Measured memory usage (in MB) for particular Events Per Second (EPS) average mes
 | **1000** | 121.6  | 126.75 | 127.94 | 140.11 | 106.82 |
 | **1500** | 128.54 | 131.9  | 137.69 | 95.21  | 113.89 |
 | **2000** | 130.62 | 125.27 | 144.59 | 98.62  | 134.61 |
+
+## Fine Tuning
```
Comment (sumo-drosiek) [inline docs/Performance.md:73]: Not sure if this should be he same header level like `logs`
Maye we should put it in separate file, IDK

**Review by kkujawa-sumo — APPROVED** [review]

File: docs/Performance.md (hunk 1 above)
Comment (pmalek-sumo) [reply docs/Performance.md:73]: We should definitely reorganize these to put test results in under one `##` section and fine tuning in a different `##` section, IMHO.

**Review by sumo-drosiek — CHANGES_REQUESTED** [review]
Lets resolve discussion regarding structure of the document

File: docs/Performance.md [hunk 2]
```diff
@@ -69,3 +84,41 @@ Measured memory usage (in MB) for particular Events Per Second (EPS) average mes
...
+
+Read more about these features in the [Batch Processor docs](https://github.com/open-telemetry/opentelemetry-collector/blob/main/processor/batchprocessor/README.md).
+
+### Memory Limiter Processor
+
+The [memorylimiterprocessor](https://github.com/open-telemetry/opentelemetry-collector/tree/main/processor/memorylimiterprocessor)
+prevents out-of-memory crashes for the collector process
+by monitoring the amount of memory used by the collector and forcing it to lower its memory consumption.
+
+Read more about its features in the [Memory Limiter Processor docs](https://github.com/open-telemetry/opentelemetry-collector/blob/main/processor/memorylimiterprocessor/README.md).
```
Comment (pmalek-sumo) [inline docs/Performance.md:124]: One nitpick from my side would be to extract those long URLs as we did elsewhere e.g. this file in line 13

Comment (astencel-sumo) [reply docs/Performance.md:124]: Hm I don't see the point in doing that, it just creates redundancy in my opinion..

Comment (astencel-sumo) [reply docs/Performance.md:128]: But OK let's do it and let it sink in for some time ;)

File: docs/Performance.md [hunk 3]
```diff
@@ -69,3 +84,44 @@ Measured memory usage (in MB) for particular Events Per Second (EPS) average mes
...
+
+[batchprocessor]: https://github.com/open-telemetry/opentelemetry-collector/tree/main/processor/batchprocessor
+
+It has the following features that can help with performance:
+
+- `send_batch_size`,
+- `send_batch_max_size`,
+- `timeout`.
+
+Read more about these features in the [Batch Processor docs](batchprocessor_docs).
```
Comment (pmalek-sumo) [inline docs/Performance.md:117]:
```suggestion
Read more about these features in the [Batch Processor docs][batchprocessor_docs].
```

File: docs/Performance.md [hunk 4]
```diff
@@ -69,3 +84,47 @@ Measured memory usage (in MB) for particular Events Per Second (EPS) average mes
...
+### Memory Limiter Processor
+
+The [memorylimiterprocessor] prevents out-of-memory crashes for the collector process
+by monitoring the amount of memory used by the collector and forcing it to lower its memory consumption.
+
+[memorylimiterprocessor]: (https://github.com/open-telemetry/opentelemetry-collector/tree/main/processor/memorylimiterprocessor)
+
+Read more about its features in the [Memory Limiter Processor docs].
+
+[Memory Limiter Processor docs]: https://github.com/open-telemetry/opentelemetry-collector/blob/main/processor/memorylimiterprocessor/README.md
```
Comment (pmalek-sumo) [inline docs/Performance.md:130]:
```suggestion
The [`memorylimiterprocessor`] prevents out-of-memory crashes for the collector process
by monitoring the amount of memory used by the collector and forcing it to lower its memory consumption.

Read more about its features in the [Memory Limiter Processor docs].

[memorylimiterprocessor]: (https://github.com/open-telemetry/opentelemetry-collector/tree/main/processor/memorylimiterprocessor)
[Memory Limiter Processor docs]: https://github.com/open-telemetry/opentelemetry-collector/blob/main/processor/memorylimiterprocessor/README.md
```

File: docs/Performance.md [hunk 5]
```diff
@@ -69,3 +84,47 @@ Measured memory usage (in MB) for particular Events Per Second (EPS) average mes
...
+
+- `retry_on_failure` with its `initial_interval`, `max_interval` and `max_elapsed_time` settings,
+- `sending_queue` with its `num_consumers`, `queue_size` settings,
+- `timeout`.
+
+Read more about these features in the [Sumo Logic Exporter docs](../pkg/exporter/sumologicexporter/README.md).
+
+### Batch Processor
+
+The [batchprocessor] joins records of each type in batches.
```
Comment (pmalek-sumo) [inline docs/Performance.md:107]:
```suggestion
The [`batchprocessor`] joins records of each type in batches.
```

**Review by pmalek-sumo — APPROVED** [review]

**Review by sumo-drosiek — APPROVED** [review]

**Pull request merged**